rholang/src/main/k/rholang.k

```diff
@@ -313,9 +320,83 @@ rule <thread>
 
 
 // ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* //
-// PERSISTENT SENDING (AND RECEIVING)
+// PERSISTENT SENDING AND RECEIVING
 // ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* //
 
+// 1. Persistent sending
+// We create an "out" cell just as we would have previously.
+rule <thread>
+        (<k> C:Chan!!(P:HigherProcs) </k> => <out> C!!(P) </out>)
+        (.Bag => <OutData> I ; L </OutData> )
+     </thread>
+     <NewIntGenerator> I => I +Int 1 </NewIntGenerator>
+     <GlobalListofOutIds> ... .List => ListItem(I) ... </GlobalListofOutIds>
+     <GlobalListofInIds> L </GlobalListofInIds> [outcreation]
+
+// Verify the ID of one is in the list of the other
+// If they are, begin checking if there is a match.
+// Do this without absorbing the "in" cell.
+
+// (a) the "in"'s ID is in the "send"'s list of receives
```

Comment:
"send's list of receives" and "receive's list of sends" onl ine 359 are hard to parse.  Can you restate these?